pkg/agent/directive_handler.go

```diff
@@ -99,7 +99,10 @@ func (dh *BasicDirectiveHandler) reportCommandResult(ctx context.Context, comman
 		result.Success = false
 		result.Message = cmdError.Error()
 	}
-	err := a.comm.ReportCommandResult(ctx, result)
+	a.ambassadorAPIKeyMutex.Lock()
+	apiKey := a.ambassadorAPIKey
+	a.ambassadorAPIKeyMutex.Unlock()
```

Comment:
Might be worth a comment that you really don't want to hold this lock during the report-command-result call.

Reply:
@kflynn Why? Where would you recommend to use the mutex to access the api key from the agent and send it down to this point?